expyfun/_eyelink_controller.py

```diff
@@ -528,30 +528,102 @@ def wait_for_fix(self, fix_pos, fix_time=0., tol=100., max_wait=np.inf,
+        -------
+        fix_success : bool
+            Whether or not the subject successfully fixated throughout the
+            `check_duration`.
+        """
+        fix_success = True
+
+        time_start = time.time()
+        time_end = time_start + check_duration
+        check_len = int(period / check_interval)
+
+        fix_pos = np.array(fix_pos)
+        if not (fix_pos.ndim == 1 and fix_pos.size == 2):
+            raise ValueError('fix_pos must be a 2-element array-like vector')
+        fix_pos = self._ec._convert_units(fix_pos[:, np.newaxis], units, 'pix')
+        fix_pos = fix_pos[:, 0]
+        check = []
+        while ((fix_success and time.time() < time_end) if stop_early else
+               time.time() < time_end):
+            if fix_success:
+                # sample eye position
+                eye_pos = self.get_eye_position()  # in pixels
+                if _within_distance(eye_pos, fix_pos, tol):
+                    check.append(True)
+                else:
+                    check.append(False)
+                fix_success = any(check[-check_len:])
+                if len(check) >= check_len:
+                    self._ec._response_handler.check_force_quit()
+            self._ec.wait_secs(check_interval)
+        return fix_success
+
     def custom_calibration(self, ctype='HV5', horiz=2./3., vert=2./3.,
-                           units='norm'):
+                           coordinates=None, units='norm'):
         """Set Eyetracker to use a custom calibration sequence
 
         Parameters
         ----------
         ctype : str
             Type of calibration. Currently 'H3', 'HV5', 'HV9', and 'HV13'
-            are supported.
+            are supported. If 'custom' coordinates can be directly input.
         horiz : float
             Horizontal distance (left and right, each) to use.
         vert : float
             Vertical distance (up and down, each) to use.
+        coordinates : list of list of float or int
+            List of coordinates to use for calibration. Each entry in the list
+            must be a list with 2 elements corresponding to x and y position.
         units : str
             Units to use. See ``check_units`` for options.
 
         See Also
         --------
         EyelinkController.calibrate
         """
-        allowed_types = ['H3', 'HV5', 'HV9', 'HV13']
+        allowed_types = ['H3', 'HV5', 'HV9', 'HV13', 'custom']
         if ctype not in allowed_types:
             raise ValueError('ctype cannot be "{0}", but must be one of {1}'
                              ''.format(ctype, allowed_types))
+        if ctype != 'custom':
+            if coordinates is not None:
+                raise ValueError('If ctype is not \'custom\' coordinates canno'
+                                 't be used to generate calibration pattern.')
+        else:
+            if not isinstance(coordinates, list):
+                raise ValueError('Coordinates must be in a list.')
```

Comment:
you don't need this check, `np.array` is good enough (you should really allow list, tuple, ndarray, etc.)

Reply:
(also this is redundant with the check that is done later)